[CI] SmokeTestMultiNodeClientYamlTestSuiteIT test {yaml=search/500_date_range/test match query gte and lte with year uuuu} failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/ctye33jrkiiv6/tests/:qa:smoke-test-multinode:yamlRestTest/org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT/test%20%7Byaml=search%2F500_date_range%2Ftest%20match%20query%20gte%20and%20lte%20with%20year%20uuuu%7D

**Reproduction line:**
`./gradlew ':qa:smoke-test-multinode:yamlRestTest' --tests "org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT.test {yaml=search/500_date_range/test match query gte and lte with year uuuu}" -Dtests.seed=D078FD3E4CFBBC93 -Dtests.locale=hi-IN -Dtests.timezone=MST -Druntime.java=17`

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT&tests.test=test%20%7Byaml%3Dsearch/500_date_range/test%20match%20query%20gte%20and%20lte%20with%20year%20uuuu%7D

**Failure excerpt:**
```
java.lang.AssertionError: Failure at [search/500_date_range:103]: 
Expected: "first"
     but: was "fifth"

  at __randomizedtesting.SeedInfo.seed([D078FD3E4CFBBC93:582CC2E4E207D16B]:0)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:510)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:479)
  at jdk.internal.reflect.GeneratedMethodAccessor11.invoke(null:-1)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

  Caused by: java.lang.AssertionError: 
  Expected: "first"
       but: was "fifth"

    at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
    at org.junit.Assert.assertThat(Assert.java:956)
    at org.junit.Assert.assertThat(Assert.java:923)
    at org.elasticsearch.test.rest.yaml.section.MatchAssertion.doAssert(MatchAssertion.java:99)
    at org.elasticsearch.test.rest.yaml.section.Assertion.execute(Assertion.java:65)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:499)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:479)
    at jdk.internal.reflect.GeneratedMethodAccessor11.invoke(null:-1)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:568)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
    at java.lang.Thread.run(Thread.java:833)

```

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/500_date_range.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/500_date_range.yml
@@ -1,4 +1,7 @@
 setup:
+  - skip:
+      version: " - 8.5.0"
+      reason: fixed in 8.5.1
   - do:
       indices.create:
         index: dates_year_only

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/500_date_range.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/500_date_range.yml
@@ -1,7 +1,4 @@
 setup:
-  - skip:
-      version: " - 8.5.99"
-      reason: awaits backports
   - do:
       indices.create:
         index: dates_year_only
@@ -16,14 +13,14 @@ setup:
       bulk:
         refresh: true
         body:
-          - '{ "index" : { "_index" : "dates_year_only", "_id" : "first" } }'
+          - '{ "index" : { "_index" : "dates_year_only", "_id" : "1" } }'
           - '{"date" : "1900", "field" : 1 }'
-          - '{ "index" : { "_index" : "dates_year_only", "_id" : "second" } }'
-          - '{"date" : "2022", "field" : 1 }'
-          - '{ "index" : { "_index" : "dates_year_only", "_id" : "third" } }'
+          - '{ "index" : { "_index" : "dates_year_only", "_id" : "2" } }'
           - '{"date" : "2022", "field" : 2 }'
-          - '{ "index" : { "_index" : "dates_year_only", "_id" : "fourth" } }'
-          - '{"date" : "1500", "field" : 2 }'
+          - '{ "index" : { "_index" : "dates_year_only", "_id" : "3" } }'
+          - '{"date" : "2022", "field" : 3 }'
+          - '{ "index" : { "_index" : "dates_year_only", "_id" : "4" } }'
+          - '{"date" : "1500", "field" : 4 }'
 
   - do:
       indices.create:
@@ -38,16 +35,16 @@ setup:
       bulk:
         refresh: true
         body:
-          - '{ "index" : { "_index" : "dates", "_id" : "first" } }'
+          - '{ "index" : { "_index" : "dates", "_id" : "1" } }'
           - '{"date" : "1900-01-01T12:12:12.123456789Z", "field" : 1 }'
-          - '{ "index" : { "_index" : "dates", "_id" : "second" } }'
-          - '{"date" : "2022-01-01T12:12:12.123456789Z", "field" : 1 }'
-          - '{ "index" : { "_index" : "dates", "_id" : "third" } }'
-          - '{"date" : "2022-01-03T12:12:12.123456789Z", "field" : 2 }'
-          - '{ "index" : { "_index" : "dates", "_id" : "fourth" } }'
-          - '{"date" : "1500-01-01T12:12:12.123456789Z", "field" : 2 }'
-          - '{ "index" : { "_index" : "dates", "_id" : "fifth" } }'
-          - '{"date" : "1500-01-05T12:12:12.123456789Z", "field" : 2 }'
+          - '{ "index" : { "_index" : "dates", "_id" : "2" } }'
+          - '{"date" : "2022-01-01T12:12:12.123456789Z", "field" : 2 }'
+          - '{ "index" : { "_index" : "dates", "_id" : "3" } }'
+          - '{"date" : "2022-01-03T12:12:12.123456789Z", "field" : 3 }'
+          - '{ "index" : { "_index" : "dates", "_id" : "4" } }'
+          - '{"date" : "1500-01-01T12:12:12.123456789Z", "field" : 4 }'
+          - '{ "index" : { "_index" : "dates", "_id" : "5" } }'
+          - '{"date" : "1500-01-05T12:12:12.123456789Z", "field" : 5 }'
 
 ---
 "test range query for all docs with year uuuu":
@@ -56,6 +53,7 @@ setup:
         rest_total_hits_as_int: true
         index: dates
         body:
+          sort: field
           query:
             range:
               date:
@@ -73,6 +71,7 @@ setup:
         rest_total_hits_as_int: true
         index: dates
         body:
+          sort: field
           query:
             range:
               date:
@@ -82,7 +81,7 @@ setup:
 
   - match: { hits.total: 1 }
   - length: { hits.hits: 1 }
-  - match: { hits.hits.0._id: "fourth" }
+  - match: { hits.hits.0._id: "4" }
 
 ---
 "test match query gte and lte with year uuuu":
@@ -91,6 +90,7 @@ setup:
         rest_total_hits_as_int: true
         index: dates
         body:
+          sort: field
           query:
             range:
               date:
@@ -100,9 +100,9 @@ setup:
 
   - match: { hits.total: 3 }
   - length: { hits.hits: 3 }
-  - match: { hits.hits.0._id: "first" }
-  - match: { hits.hits.1._id: "fourth" }
-  - match: { hits.hits.2._id: "fifth" }
+  - match: { hits.hits.0._id: "1" }
+  - match: { hits.hits.1._id: "4" }
+  - match: { hits.hits.2._id: "5" }
 
 ---
 "test match query with year uuuu":
@@ -111,6 +111,7 @@ setup:
         rest_total_hits_as_int: true
         index: dates_year_only
         body:
+          sort: field
           query:
             match:
               date:
@@ -118,4 +119,4 @@ setup:
 
   - match: { hits.total: 1 }
   - length: { hits.hits: 1 }
-  - match: { hits.hits.0._id: "fourth" }
+  - match: { hits.hits.0._id: "4" }

